### PR TITLE
Adds negative damage scaling to Buckshot.

### PR DESCRIPTION
**Pull request by LanCartwright**
:cl: 
add: Adds scaling damage to buckshot.
/:cl:

[]: # 
Buckshot now deals 0.75 damage less per tile travelled.
This weakens buckshot which was previously had it's pellets buffed.

This is subject to change.

### CODE REVIEW

**Comment by kevinz000** [comment]
Does this heal you if you go beyond 20 tiles ?

**Comment by LanCartwright** [comment]
oh shit. I did not factor that potential in. Let me do a quick fix.

pellets now qdel(src) after breaking 0 damage threshold.

**Comment by ExcessiveUseOfCobblestone** [comment]
that's.... that's actually a funny feature.

Shame on you mekhi!

**Comment by Robustin** [comment]
Long time coming, people are really going to notice when clicking 3 times down a hallway doesnt everyone stunned/crit.

**Comment by optimumtact** [comment]
This doesn't help because the shotgun is really only strongest at close range anyway due to the pellet spread.

**Comment by ExcessiveUseOfCobblestone** [comment]
as it should be and not act as a hallway sniper

**Comment by Haevacht** [comment]
Put the force check on impact. If force <0 then return.

**Comment by Steelpoint** [comment]
So a shotgun shell has, last I checked, about six pellets pet shell. At 12.5 damage each that's a maximum amount of 75 damage at point blank range, meaning right next to a target.

So pellets go like this.

random

Once your past about three tiles (11 on image above) you usually only hit two or one pellets on a target, at three tiles you typically hit three or four.

It seems fine at least, though the shotty shells should deal a bit more damage now, considering their damage was nerfed to make them less effective at range.

File: code/modules/projectiles/projectile/bullets.dm [hunk 1]
```diff
@@ -35,6 +35,12 @@
 	name = "pellet"
 	damage = 12.5
 
+/obj/item/projectile/bullet/pellet/Range()
+	..()
+	damage += -0.75
```
Comment (Iamgoofball) [inline code/modules/projectiles/projectile/bullets.dm]: just do damage -= 0.75 ffs

File: code/modules/projectiles/projectile/bullets.dm [hunk 2]
```diff
@@ -35,6 +35,12 @@
 	name = "pellet"
 	damage = 12.5
 
+/obj/item/projectile/bullet/pellet/Range()
+	..()
+	damage += -0.75
+	if(damage < 0)
```
Comment (Iamgoofball) [inline code/modules/projectiles/projectile/bullets.dm]: what if it's equal to 0

Comment (PKPenguin321) [inline code/modules/projectiles/projectile/bullets.dm]: then if it hits you it just does nothing

**Comment by anconfuzedrock** [comment]
if people can't spam their 75 damage max pellets down a hallway, they are just going to shoot 3+ slugs down it, and those do 60. rip buckshot 2017 you were more fair than not but every lethal ranged weapon sucks balls stuns only

**Comment by LanCartwright** [comment]
regarding any thoughts on qdelling the projectile, I could do this to preserve any immersion concerns:
```
if(damage <0)
 new/obj/item/trash/pellet(src.loc)
 qdel(src)
```

**Comment by ExcessiveUseOfCobblestone** [comment]
@anconfuzedrock slugs should be less pellets then :^)

**Comment by Kromgar** [comment]
Nerftheslugs2017

**Comment by Iamgoofball** [comment]
Remember during goofguns we figured out slugs are revolvers

**Comment by LanCartwright** [comment]
I'd like a review. 
Do you want anything changed?
Would you like me to implement physical bullet drop-off? 
Do you want damage falloff for all projectiles?

**Comment by Ninja137** [comment]
Falloff damage for lasers would probably be a good idea. Hell, make them do less damage when they travel through smoke and shit, too.

**Pull request merged**